packages/components/src/components/as-histogram-widget/utils/brush.service.ts

```diff
@@ -0,0 +1,95 @@
+import { BrushBehavior, brushX } from 'd3-brush';
+import { SVGContainer, SVGGContainer } from '../types/Container';
+
+export function addBrush(
+  width: number,
+  height: number,
+  onBrush: () => void,
+  onBrushEnd: () => void,
+  CUSTOM_HANDLE_WIDTH: number,
+  CUSTOM_HANDLE_HEIGHT,
+  X_PADDING: number,
+  Y_PADDING: number): BrushBehavior<{}> {
+
+  return brushX()
+    .handleSize(CUSTOM_HANDLE_WIDTH)
+    .extent([[0, 0], [width - X_PADDING, height - Y_PADDING + (CUSTOM_HANDLE_HEIGHT / 2)]])
+    .on('brush', onBrush)
+    .on('end', onBrushEnd);
+}
+
+export function addBrushArea(
+  brush: BrushBehavior<{}>,
+  container: SVGContainer,
+): SVGGContainer {
+
+  let brushArea = container.select<SVGGElement>('g.brush');
+
+  if (brushArea.empty()) {
+    brushArea = container.append('g');
+    brushArea.attr('class', 'brush');
+  }
+
+  brushArea.call(brush);
+
+  return brushArea;
+}
+
+export function addCustomHandles(
+  brushArea: SVGGContainer,
+  height: number,
+  CUSTOM_HANDLE_WIDTH: number,
+  CUSTOM_HANDLE_HEIGHT: number,
+  Y_PADDING: number
+): SVGGContainer<{ type: string }, SVGGElement, {}> {
+
+  let bottomLine = brushArea.select('line.bottomline');
+
+  if (bottomLine.empty()) {
+    bottomLine = brushArea.append('line')
+      .attr('class', 'bottomline')
+      .attr('stroke-width', 4)
+      .style('opacity', 0)
+      .attr('pointer-events', 'none');
+  }
+
+  bottomLine
+    .attr('y1', height - Y_PADDING)
+    .attr('y2', height - Y_PADDING);
+
+  let customHandlers = brushArea.selectAll<SVGGElement, { type: string }>('.handle--wrapper');
```

Comment:
Handlers or handles?

Reply:
It's handles, I noticed this but totally forgot to change it.